dm/dm/master/scheduler/scheduler.go

```diff
@@ -1039,6 +1034,73 @@ func (s *Scheduler) RemoveSubTasks(task string, sources ...string) error {
 	return nil
 }
 
+// UpdateSubTasks update the information of one or more subtasks for one task.
+func (s *Scheduler) UpdateSubTasks(ctx context.Context, cfgs ...config.SubTaskConfig) error {
+	if !s.started.Load() {
+		return terror.ErrSchedulerNotStarted.Generate()
+	}
+	if len(cfgs) == 0 {
+		return nil // no subtasks need to add, this should not happen.
+	}
+	taskNamesM := make(map[string]struct{}, 1)
+	for _, cfg := range cfgs {
+		taskNamesM[cfg.Name] = struct{}{}
+	}
+	taskNames := strMapToSlice(taskNamesM)
```

Comment:
we can directly call len() on map

Reply:
need taskNames to generate terror in L 1054